src/models/event.js

```diff
@@ -3,14 +3,67 @@ import Attributes from './attributes';
 import EventParticipant from './event-participant';
 
 export default class Event extends RestfulModel {
+  
+  get start() {
+    return 
+      this.when.start_time ||
+      this.when.start_date ||
+      this.when.time ||
+      this.when.date;
+  }
+
+  get end() {
+    return
+      this.when.end_time ||
+      this.when.end_date ||
+      this.when.time ||
+      this.when.date;
+  }
+
+  set start(val) {
+    if (this.when) {
+      if (typeof val === 'number') {
+        if (val === this.when.end_time) {
+          this.when = { time: val }
+        } else {
+          delete this.when.time;
```

Comment:
should we also proactively delete `when.start_date` and `when.date`, just in case?

Reply:
good point